serve.go

```diff
@@ -22,15 +22,21 @@ func (s *Server) Run() error {
 	if s.corsMiddleware != nil {
 		s.Server.Handler = s.corsMiddleware(s.Server.Handler)
 	}
-
+	if s.isTLS() {
```

Comment:
The issue with this approach is that it's very easy for the user to misconfigure something. We should provide a simple way of starting the server that is similar to the stdlib.

For users that need more the WithTLS config is fine, but more work is needed to support that a bit better (hence why it should be it's own PR). With the RunTLS approach we cover a large portion of use cases. For users that need to get into the nuts and bolts of a TLSConfig they can still do that as `Server` is exposed. And they can simply call `app.RunTLS("","")`.

Reply:
Now that I have real tests to check all of this I understand what you mean. Adding some errors in `Run` to account for invalid configuration didn't seem like a big deal but with certificates being provided by `tls.Config.Certificates` or `tls.Config.GetCertificate` or files for ListenAndRunTLS is not as simple as I initially though. 

I'm afraid that once we added `RunTLS` that would be in there forever, I would really like to have that magic Run option but for now I've followed your wise advice and simplified things.